velox/common/caching/tests/AsyncDataCacheTest.cpp

```diff
@@ -1438,7 +1438,7 @@ TEST_P(AsyncDataCacheTest, ssdWriteOptions) {
       {0.8, 0.95, 4UL << 20, false},
       {0.8, 0.3, 32UL << 20, false},
       {0.8, 0.3, 4UL << 20, true},
-      {0.001, 0.3, 4UL << 20, true}};
+      {0.0, 1, 4UL << 20, true}};
```

Comment:
I might be off context. Correct me if I'm wrong, but I'm guessing this test is flaky because when write ratio is super small it is still not guaranteed to be flushed to ssd due to the loose ssdsavable ratio. If that is the case, then I am thinking maybe we can keep other things unchanged and make ssd savable ratio super small to make sure the first entry is always flushed/written to ssd so that the EQ check is always satisfied.

Reply:
In some very rare corner cases, I've seen ssd write was not triggered on all the shards.

Reply:
Actually I don't know the exact reason of the flakiness as it's very hard to repro (around 1 failure out of 10K stress runs). This change is based on assumptions:
1. `maxWriteRatio` is small but not zero, which means if the entry size is small enough they can still write more than 1 entry.
2. `ssdSavableRatio` as 0.3 and `minSsdSavableBytes` as 4MB cannot guarantee the ssd write will only be triggered once while making it 100% can guarantee that in theory: `memory/(memory + ssd)` will be 100% only when ssd cache is zero. Once ssd cache > 0, it will never be triggered again.